kore/src/Kore/Exec.hs

```diff
@@ -364,12 +368,20 @@ exec
         transit ::
             GraphTraversal.TState
                 Prim
-                (ExecDepth, ProgramState (Pattern RewritingVariableName)) ->
+                ( ExecDepth
+                , ProgramState
+                    (Predicate RewritingVariableName, Substitution RewritingVariableName, UniqueId)
```

Comment:
These triplets keep coming up, maybe we should define a dedicated type? 
```
data RuleInfo a =
    RuleInfo 
        { rulePredicate :: Predicate a
        , ruleSubstitution :: Substitution a
        , ruleId :: UniqueId 
        }
```
Not sure where it would live, whether in here or in `Kore.Rewrite` .